build.go

```diff
@@ -10,16 +10,21 @@ import (
 	"path/filepath"
 )
 
-func build(token, configRoot, workspaceRoot, clonePath, owner, repo, ref string, extraEnv []string) (*executeResult, string, error) {
+func build(token, configRoot, workspaceRoot, resultRoot, clonePath, owner, repo, ref string, extraEnv []string) (*executeResult, string, string, error) {
 
 	workspacePath, err := prepareWorkspace(token, workspaceRoot, clonePath, owner, repo, ref)
 	if err != nil {
-		return nil, workspacePath, fmt.Errorf("failed to prepare workspace: %v", err)
+		return nil, workspacePath, "", fmt.Errorf("failed to prepare workspace: %v", err)
+	}
+
+	resultRootPath, err := findLogDirPath(resultRoot, owner, repo)
```

Comment:
Actually looking at this, the logDIrPath is not used internal to this function.  So why are you getting it here?  Its causing the function to have more return values than it needs.